### PR TITLE
Feature/application endpoints

### PR DESCRIPTION
**Pull request by ingridarev**

### CODE REVIEW

File: project/src/main/java/com/carlease/project/application/Application.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import lombok.Data;
+
+import java.math.BigDecimal;
+import java.util.Date;
+
+@Entity
+@Table(name="applications")
+@Data
+public class Application {
+
```
Comment (Teanquisitor) [inline project/src/main/java/com/carlease/project/application/Application.java:18]: extra blank line

File: project/src/main/java/com/carlease/project/application/ApplicationRepository.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.carlease.project.application;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+
```
Comment (Teanquisitor) [inline project/src/main/java/com/carlease/project/application/ApplicationRepository.java:4]: blank line

**Review by Teanquisitor — APPROVED** [review]

File: desktop.ini [hunk 3]
```diff
@@ -0,0 +1,4 @@
+[ViewState]
```
Comment (grkarolis) [inline desktop.ini:1]: What are these for?

File: project/src/main/java/com/carlease/project/application/Application.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    @Enumerated(EnumType.STRING)
+    private ApplicationStatus status;
+
+    @Column(name = "start_date")
+    @Temporal(TemporalType.DATE)
+    private Date startDate;
+
+    @Column(name = "end_date")
+    @Temporal(TemporalType.DATE)
+    private Date endDate;
```
Comment (grkarolis) [inline project/src/main/java/com/carlease/project/application/Application.java:69]: Use LocalDate

File: project/src/main/java/com/carlease/project/application/Application.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+
+    @Column(name = "application_id")
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long applicationId;
+
+    @ManyToOne
+    @JoinColumn(name = "user_id")
+    private User user;
```
Comment (grkarolis) [inline project/src/main/java/com/carlease/project/application/Application.java:27]: How will you know which user is the one who submitted application, who is the reviewer and who is the approver?

File: project/src/main/java/com/carlease/project/application/ApplicationController.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+    @GetMapping(produces = "application/json")
+    ResponseEntity<List<Application>> getAllApplications() {
+        List<Application> list = applicationService.findAll();
+        return new ResponseEntity<>(list, HttpStatus.OK);
+    }
+
+    @GetMapping("/{id}")
+    ResponseEntity<Application> getApplicationById(@PathVariable("id") long id) {
+
```
Comment (grkarolis) [inline project/src/main/java/com/carlease/project/application/ApplicationController.java:30]: empty line

File: project/src/main/java/com/carlease/project/application/ApplicationController.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+public class ApplicationController {
+
+    public ApplicationService applicationService;
+
+    public ApplicationController(ApplicationService applicationService) {
+        this.applicationService = applicationService;
+    }
+
+    @GetMapping(produces = "application/json")
+    ResponseEntity<List<Application>> getAllApplications() {
```
Comment (grkarolis) [inline project/src/main/java/com/carlease/project/application/ApplicationController.java:23]: getApplications

File: project/src/main/java/com/carlease/project/application/ApplicationController.java [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+        List<Application> list = applicationService.findAll();
+        return new ResponseEntity<>(list, HttpStatus.OK);
+    }
+
+    @GetMapping("/{id}")
+    ResponseEntity<Application> getApplicationById(@PathVariable("id") long id) {
```
Comment (grkarolis) [inline project/src/main/java/com/carlease/project/application/ApplicationController.java:29]: getApplication

File: project/src/main/java/com/carlease/project/application/ApplicationController.java [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+        Application application = applicationService.findById(id);
+        return new ResponseEntity<>(application, HttpStatus.OK);
+    }
+
+    @PostMapping
+    ResponseEntity<Application> createApplication(@RequestBody Application application) {
```
Comment (grkarolis) [inline project/src/main/java/com/carlease/project/application/ApplicationController.java:36]: Using same class for persistance and API contract request/response is a bad practice. You must use separate entities for persistance and separate for API contract

File: project/src/main/java/com/carlease/project/application/ApplicationRepository.java [hunk 10]
```diff
@@ -0,0 +1,8 @@
...
+import java.util.List;
+
```
Comment (grkarolis) [inline project/src/main/java/com/carlease/project/application/ApplicationRepository.java:6]: @Repository

**Pull request merged**